client/js/models/archive.js

```diff
@@ -241,30 +241,49 @@ module.exports = {
       var readStream = archive.createFileReadStream(data.entryName)
       done(readStream)
     },
-    download: function (data, state, send, done) {
-      // XXX: TODO: failover msg when dat is shared from cli, no file contents available
+    downloadAsZip: function (data, state, send, done) {
       const archive = state.instance
       const zip = new Jszip()
       var zipName
       if (data && data.entryName) {
-        zipName = data.entryName
-        zip.file(data.entryName, archive.createFileReadStream(data.entryName))
+        archive.get(data.entryName, {timeout: 1000}, function (err, entry) {
+          if (err) return failoverMsg()
+          if (!archive.isEntryDownloaded(entry)) return failoverMsg()
```

Comment:
i think you have to run `archive.download(entry, function (err) { ... }` or `createFileReadStream` first to trigger a download, otherwise isEntryDownloaded(entry) will return false with sparse mode on. @clkao